src/zenml/models/component_models.py

```diff
@@ -126,6 +130,10 @@ class ComponentFilterModel(ShareableWorkspaceScopedFilterModel):
         default=None, description="Workspace of the stack component"
     )
     user_id: Union[UUID, str] = Field(None, description="User of the stack")
+    metadata_values: bytes = Field(
```

Comment:
I think having the metadata as `bytes` on the filter model makes it almost unusable?

I would assume I need to pass in the same bytes of all the component metadata to make it work as a filter? Much more useful would be if I could just check for a single (or multiple) key-value pairs, e.g. if the metadata attribute would also be `Dict[str, str]` here. The filter logic implementation would be much more complicated though, but useful to have also for future features.

Reply:
You're right, having the metdata as bytes doesn't allow filtering. I agree that this should be a Dict instead; with logic implemented for filtering. For now, I have removed the option to filter by metadata altogether and will revisit this problem if I have time.